### PR TITLE
Factor out file creation for SPO

### PR DESCRIPTION
**Pull request by Jimbo4350**
# Description

Add your description here, if it fixes a particular issue please provide a
[link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=)
to the issue.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: cardano-testnet/src/Testnet/Components/Configuration.hs [hunk 1]
```diff
@@ -58,6 +70,81 @@ createConfigYaml tempAbsPath anyCardanoEra' = GHC.withFrozenCallStack $ do
...
+      createStakedInitialGenesisFile = tempAbsPath' </> "genesis.spec.json"
+
+  -- TODO: We need to read the genesis files into Haskell and modify them
+  -- based on cardano-testnet's cli parameters
+
+  -- We create the initial genesis file to avoid having to re-write the genesis file later
+  -- with the parameters we want. The user must provide genesis files or we will use a default.
+  -- We should *never* be modifying the genesis file after cardano-testnet is run because this
+  -- is sure to be a source of confusion if users provide genesis files and we are mutating them
+  -- without their knowledge.
```
Comment (newhoggy) [inline cardano-testnet/src/Testnet/Components/Configuration.hs:93]: Good warning!

File: cardano-testnet/src/Testnet/Start/Cardano.hs [hunk 2]
```diff
@@ -90,309 +88,261 @@ cardanoTestnet testnetOptions Conf {tempAbsPath} = do
...
+    H.failMessage GHC.callStack "Specifying node configuration files per node not supported yet."
+  else do
+    startTime <- H.noteShow $ DTC.addUTCTime startTimeOffsetSeconds currentTime
+
+    H.createDirectoryIfMissing_ logDir
+
+    H.lbsWriteFile (tempAbsPath' </> "byron.genesis.spec.json")
+      . J.encode $ defaultByronProtocolParamsJsonValue
+
+    -- stuff
```
Comment (newhoggy) [inline cardano-testnet/src/Testnet/Start/Cardano.hs:110]: Stray comment `stuff`?

File: cardano-testnet/src/Testnet/Start/Cardano.hs [hunk 3]
```diff
@@ -90,309 +88,261 @@ cardanoTestnet testnetOptions Conf {tempAbsPath} = do
...
+    Byron.createByronGenesis
+      testnetMagic
+      startTime
+      Byron.byronDefaultGenesisOptions
+      (tempAbsPath' </> "byron.genesis.spec.json")
+      (tempAbsPath' </> "byron-gen-command")
+    -- Because in Conway the overlay schedule and decentralization parameter
+    -- are deprecated, we must use the "create-staked" cli command to create
+    -- SPOs in the ShelleyGenesis
```
Comment (newhoggy) [inline cardano-testnet/src/Testnet/Start/Cardano.hs:119]: Is this comment describing the code above or below?

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs [hunk 4]
```diff
@@ -71,7 +71,7 @@ hprop_leadershipSchedule = H.integrationRetryWorkspace 2 "babbage-leadership-sch
       cTestnetOptions = cardanoDefaultTestnetOptions
                           { cardanoNodes = cardanoDefaultTestnetNodeOptions
                           , cardanoEpochLength = 1000
-                          , cardanoSlotLength = 0.02
+                          , cardanoSlotLength = 0.1
```
Comment (newhoggy) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs:74]: Can you add an explanation in comments why this is `0.1` now?

**Review by newhoggy — APPROVED** [review]

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs (hunk 4 above)
Comment (Jimbo4350) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs:74]: we were rewriting the shelley genesis file with 0.1 so it wasn't being used. I'll leave add a comment.

Comment (Jimbo4350) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs:73]: Added a message in the commit

**Pull request merged**